module/init/socket-functions.mjs

```diff
@@ -129,6 +129,45 @@ async function socketAddChipsActor(actorId, chips) {
   _chatMessage(ChatMessage.getSpeaker(), actor.name, message);
 }
 
+async function socketAddChipsMarshal(chips) {
+  // extract the valid data from chips
+  let { white, red, blue } = chips;
+
+  white = typeof white === 'number' ? Math.floor(white) : 0;
+  red = typeof red === 'number' ? Math.floor(red) : 0;
+  blue = typeof blue === 'number' ? Math.floor(blue) : 0;
+
+  let total = white + red + blue;
+
+  // If we aren't actually adding any chips
+  if (total <= 0) return;
+
+  const adding = { white, red, blue };
+  const chatStr = Chips.makeMarshalReport('Added', adding);
+  _chatMessage(ChatMessage.getSpeaker(), 'Marshal', chatStr);
```

Comment:
This broadcasts for all players to see, did you mean to post to the marshal only?

Reply:
I did, the Marshal can see his own chip pot and doesn't really need a report that he's done the thing he just did. We could possibly add a setting to turn it off.

Reply:
yes if we can't get it constraining messages so only the marshal can see them we should just stop broadcasting this.